docs(react-native): async view model instance creation via useViewModelInstance({ async: true }) (#797)
rive-app/rive-docs: runtimes/react-native/data-binding.mdx

    const { instance, isLoading, error } = useViewModelInstance(riveFile, { async: true, instanceName: 'PersonInstance' });
    const { instance, isLoading, error } = useViewModelInstance(riveFile, { async: true, viewModelName: 'Settings', instanceName: 'UserSettings' });

    // From a ViewModel object
    const viewModel = await riveFile?.viewModelByNameAsync('My View Model');
    const { instance: namedInstance, isLoading, error } = useViewModelInstance(viewModel, { async: true, name: 'My Instance' });
    const { instance: newInstance, isLoading, error } = useViewModelInstance(viewModel, { async: true, useNew: true });

    // With required: true (throws once resolved to null, use with Error Boundary)
    const { instance, isLoading, error } = useViewModelInstance(riveFile, { async: true, required: true });

    // With onInit to set initial values before the instance is exposed or bound
    const { instance, isLoading, error } = useViewModelInstance(riveFile, {
        async: true,
        onInit: (vmi) => {
            vmi.numberProperty('health')?.set(100);
        },
    });
    ```

    <Info>

>         `async: true` is available from `@rive-app/react-native@0.4.18`. It is a transitional flag: the synchronous creation path (which also blocks the JS thread during render) is being phased out in stages. Today, calls without `async: true` are deprecated in TypeScript; a future release adds a runtime deprecation warning; later, `async: true` becomes required; finally the flag is removed and the hook is always asynchronous.

Reviewer: ```suggestion
        `async: true` is available in `@rive-app/react-native@0.4.18` and later. The synchronous creation path is deprecated and will be removed in a future release.
```
